media/js/base/global.js

```diff
@@ -149,6 +137,174 @@ function isFirefoxESR(userAgent, buildID) {
+ * @return {String} version number.
+ */
+function getFirefoxVersion(ua) {
+    ua = ua || navigator.userAgent;
+
+    var matches = /Firefox\/(\d+\.\d+(?:\.\d+)?)/.exec(ua);
+
+    return (matches !== null && matches.length > 0) ? matches[1] : '0';
+};
+
+/**
+ * Get the user's Firefox major version number.
+ *
+ * @param  {String} ua - browser's user agent string, navigator.userAgent is
+ *                  used if not specified
+ * @return {Integer} major version number
+ */
+function getFirefoxMasterVersion(ua) {
+    ua = ua || navigator.userAgent;
+
+    return parseInt(getFirefoxVersion(ua), 10);
+}
+
+/**
+ * Detect whether the user's Firefox is up to date or outdated. This data is
+ * mainly used for security notifications.
+ *
+ * @param  {Boolean} strict - whether the minor and patch-level version numbers
+ *                   should be compared. Default: true
+ * @param  {Boolean} isESR - whether the Firefox update channel is ESR
+ * @param  {String}  userVer - browser's version number
+ * @return {Boolean} result
+ */
+function isFirefoxReallyUpToDate(strict, isESR, userVer) {
+    strict = strict === undefined ? true : strict;
+    isESR = isESR === undefined ? isFirefoxESR() : isESR;
+    userVer = userVer === undefined ? getFirefoxVersion() : userVer;
+
+    var $html = $(document.documentElement);
+
+    if (!$html.attr('data-esr-versions') || !$html.attr('data-latest-firefox')) {
+        return false;
+    }
+
+    var versions = isESR ? $.parseJSON($html.attr('data-esr-versions').replace(/\'/g, '"'))
+                         : [$html.attr('data-latest-firefox')];
+    var userVerArr = userVer.match(/^(\d+\.\d+(?:\.\d+)?)/)[1].split('.');
+    var isUpToDate = false;
+
+    // Compare the newer version first
+    versions.sort(function(a, b) { return parseFloat(a) < parseFloat(b); });
+
+    // Only check the major version in non-strict comparison mode
+    if (!strict) {
+        userVerArr.length = 1;
+    }
+
+    for (var i = 0; i < versions.length; i++) {
+        var latestVerArr = versions[i].split('.');
+
+        // Only check the major version in non-strict comparison mode
+        if (!strict) {
+            latestVerArr.length = 1;
+        }
+
+        for (var j = 0; j < userVerArr.length; j++) {
+            if (Number(userVerArr[j]) < Number(latestVerArr[j] || 0)) {
+                isUpToDate = false;
+                break;
+            } else {
+                isUpToDate = true;
+            }
+        }
+
+        if (isUpToDate) {
+            break;
+        }
+    }
+
+    return isUpToDate;
+};
+
+/**
+ * Use the async mozUITour API of Firefox to retrieve the user's browser info,
+ * including the update channel and accurate, patch-level version number. This
+ * API is available on Firefox 35 and later.
+ * http://bedrock.readthedocs.org/en/latest/uitour.html
+ *
+ * @param  {Function} callback - callback function to be executed
+ * @return {None}
+ */
+function getFirefoxDetails(callback) {
```

Comment:
:+1: Let's add those new functions to a new js module and cache the API's query result on it.